Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/SpatialAwarenessSystemInspector.cs

```diff
@@ -0,0 +1,136 @@
+                EditorGUILayout.Vector3Field("Extents", observer.ObservationExtents);
+
+                EditorGUILayout.EndVertical();
+                observerIndex++;
+            }
+
+            GUI.color = enabledColor;
+
+            if (observerIndex == 0)
+            {
+                EditorGUILayout.LabelField("(None found)", EditorStyles.miniLabel);
+            }
+
+            EditorGUILayout.Space();
+
+            EditorGUILayout.LabelField("Editor Options", EditorStyles.boldLabel);
+            ShowObserverBoundary = SessionState.GetBool(ShowObserverBoundaryKey, false);
+            ShowObserverBoundary = EditorGUILayout.Toggle("Show Observer Boundaries", ShowObserverBoundary);
+            SessionState.SetBool(ShowObserverBoundaryKey, ShowObserverBoundary);
+
+            ShowObserverOrigin = SessionState.GetBool(ShowObserverOriginKey, false);
+            ShowObserverOrigin = EditorGUILayout.Toggle("Show Observer Origins", ShowObserverOrigin);
+            SessionState.SetBool(ShowObserverOriginKey, ShowObserverOrigin);
+        }
+
+        public override void DrawSceneGUI(object target, SceneView sceneView) { }
+
+        public override void DrawGizmos(object target)
+        {
+            if (!(ShowObserverBoundary || ShowObserverOrigin))
+            {
+                return;
+            }
+
+            MixedRealitySpatialAwarenessSystem spatial = (MixedRealitySpatialAwarenessSystem)target;
+
+            int observerIndex = 0;
+            foreach (IMixedRealitySpatialAwarenessObserver observer in spatial.GetObservers())
+            {
+                Gizmos.color = GetObserverColor(observerIndex);
+
+                if (ShowObserverBoundary)
+                {
+                    switch (observer.ObserverVolumeType)
+                    {
+                        case VolumeType.None:
+                            break;
+
+                        case VolumeType.AxisAlignedCube:
+                            Gizmos.DrawWireCube(observer.ObserverOrigin, observer.ObservationExtents);
+                            break;
+
+                        case VolumeType.Sphere:
+                            Gizmos.DrawWireSphere(observer.ObserverOrigin, observer.ObservationExtents.x);
+                            break;
+
+                        case VolumeType.UserAlignedCube:
+                            Gizmos.DrawWireCube(observer.ObserverOrigin, observer.ObservationExtents);
+                            break;
+                    }
+                }
+
+                Gizmos.matrix = Matrix4x4.identity;
+
+                if (ShowObserverOrigin)
+                {
+                    Gizmos.DrawSphere(observer.ObserverOrigin, 0.1f);
+                }
+
+                observerIndex++;
+            }
+        }
+
+        private Color GetObserverColor(int observerIndex)
+        {
+            if (observerIndex >= observerColors.Length)
+            {
+                observerIndex -= observerColors.Length;
```

Comment:
>observerIndex -= observerColors.Length; [](start = 16, length = 39)

This doesn't work out if the observerIndex is > 2x the observerColors length.

For example, you have 13 observers, and the observerColors length is 6.

This will still try to map it to position 7, which doesn't exist.

The canonical way to do wraparound here is just modulo i.e.:

observerIndex = observerIndex % observerColors.Length;

Reply:
Not sure what I was smoking here - the line should be `observerIndex  = 0;`